scripts/src/models/orders.ts

```diff
@@ -133,6 +133,17 @@ export class Order extends CreationDateModel {
 			.execute();
 	}
 
+	public static getCompletedOrder<T extends Order>(userId: string, offerId: string): Promise<Order> {
+		const query = Order.createQueryBuilder()
+			.where(new Brackets(qb => {
+				qb.where("status = :completed", { completed: "completed" })
+					.orWhere("status = :pending", { pending: "pending" });
+			}))
+			.andWhere("offer_id = :offerId", { offerId })
+			.andWhere("user_id = :userId", { userId });
+		return query.getOne() as Promise<Order>;
```

Comment:
do you really need the type assertion here? it should know that it's `Promise<Order>`.
Also, why don't you just make it an async function?